dspy/teleprompt/bootstrap_finetune.py

```diff
@@ -175,58 +186,73 @@ def bootstrap_trace_data(
     program: Program,
     dataset: List[Example],
     metric: Optional[Callable] = None,
-    num_threads=6,
+    num_threads: int =6,
+    max_errors: int =10,
 ) -> List[Dict[str, Any]]:
     # Return a list of dicts with the following keys:
     #     example_ind, example, prediction, trace, and score (if metric != None)
     evaluator = Evaluate(
-        devset=dataset, num_threads=num_threads, display_progress=True, return_outputs=True,
-        provide_traceback=True  # TODO(check with team)
+        devset=dataset, num_threads=num_threads, display_progress=False, return_outputs=True,
+        provide_traceback=False, max_errors=max_errors
     )
 
-    def wrapped_metric(example, prediction, trace=None):
-        prediction, _ = prediction
-        return metric(example, prediction, trace) if metric else True
-
-    def wrapped_program(**kwargs):
-        with dspy.context(trace=[]):
-            return program(**kwargs), dspy.settings.trace.copy()
-
-    _, outputs = evaluator(wrapped_program, metric=wrapped_metric)
+    _metric = metric if metric else lambda example, prediction: 1
```

Comment:
Why is this part removing the code from wrapped_program? It's there for a good reason. It avoid looping and calling the program twice.